dvc/pkg.py

```diff
@@ -0,0 +1,120 @@
+import os
+import shutil
+
+from funcy import cached_property
+from schema import Optional
+
+from dvc.config import Config
+from dvc.cache import CacheConfig
+from dvc.exceptions import DvcException
+from dvc.utils.compat import urlparse
+
+
+class NotInstalledPkgError(DvcException):
+    def __init__(self, name):
+        super(NotInstalledPkgError, self).__init__(
+            "Package '{}' is not installed".format(name)
+        )
+
+
+class Pkg(object):
+    PARAM_NAME = "name"
+    PARAM_URL = "url"
+    PARAM_VERSION = "version"
+
+    SCHEMA = {
+        Optional(PARAM_NAME): str,
+        Optional(PARAM_URL): str,
+        Optional(PARAM_VERSION): str,
+    }
+
+    def __init__(self, pkg_dir, **kwargs):
+        self.pkg_dir = pkg_dir
+        self.url = kwargs.get(self.PARAM_URL)
+
+        name = kwargs.get(self.PARAM_NAME)
+        if name is None:
+            name = os.path.basename(self.url)
+        self.name = name
+
+        self.version = kwargs.get(self.PARAM_VERSION)
+        self.path = os.path.join(pkg_dir, self.name)
+
+    @cached_property
+    def repo(self):
+        from dvc.repo import Repo
+
+        if not self.installed:
+            raise NotInstalledPkgError(self.name)
+
+        return Repo(self.path)
+
+    @property
+    def installed(self):
+        return os.path.exists(self.path)
+
+    def install(self, cache_dir=None):
+        import git
+
+        if self.installed:
+            return
+
+        git.Repo.clone_from(self.url, self.path)
+
+        if self.version:
+            self.repo.scm.checkout(self.version)
```

Comment:
just a question to think about .. can we depend on multiple versions of the same package?

Reply:
Yes, it is going to be added soon. This line is for those cases when you are explicitly installing some particular version of the package and want to simply reference it as `mypkg` instead of `mypkg, version: version`. Thanks for the heads up though!

Reply:
Fixed.